sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/options/AwsModule.java

```diff
@@ -122,6 +123,8 @@ public AWSCredentialsProvider deserializeWithType(
         return new SystemPropertiesCredentialsProvider();
       } else if (typeName.equals(ProfileCredentialsProvider.class.getSimpleName())) {
         return new ProfileCredentialsProvider();
+      } else if (typeName.equals(WebIdentityTokenCredentialsProvider.class.getSimpleName())) {
+        return WebIdentityTokenCredentialsProvider.create();
```

Comment:
Hi, interestingly I wrote a similar code to this to upgrade AWS client and support Pod Identity.

However I did not have to add another "else if" case; I exploited the existing DefaultAWSCredentialsProviderChain which you can call by specifying "--awsCredentialsProvider={\"@type\": \"DefaultAWSCredentialsProviderChain\"}".

The default provider chain already incorporates the WebIdentityTokenCredentialsProvider.

See https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-core/src/main/java/com/amazonaws/auth/DefaultAWSCredentialsProviderChain.java#L48 here for the relevant line in the AWS client.

Reply:
Nice info, @andeb would you be up to do the minor cleanup?

Reply:
yup will do @iemejia

Reply:
@iemejia @andeb opened PR https://github.com/apache/beam/pull/10836 as discussed.